MapboxNavigation/NavigationViewController.swift

```diff
@@ -641,12 +602,15 @@ extension NavigationViewController: NavigationServiceDelegate {
         delegate?.navigationViewController?(self, willRerouteFrom: location)
     }
     
-    @objc public func navigationService(_ service: NavigationService, didRerouteAlong route: Route, at: CLLocation?, proactive: Bool) {
-        mapViewController?.notifyDidReroute(route: route)
+    @objc public func navigationService(_ service: NavigationService, didRerouteAlong route: Route, at location: CLLocation?, proactive: Bool) {
+        navigationComponents.forEach { $0.navigationService?(service, didRerouteAlong: route, at: location, proactive: proactive) }
+
         delegate?.navigationViewController?(self, didRerouteAlong: route)
     }
     
     @objc public func navigationService(_ service: NavigationService, didFailToRerouteWith error: Error) {
+        navigationComponents.forEach { $0.navigationService?(service, didFailToRerouteWith: error) }
```

Comment:
Functionally equivalent but clearer:

```suggestion
        for component in navigationComponents {
            component.navigationService?(service, didFailToRerouteWith: error)
        }
```